### ISSUE TITLE
iteration for two types of residue rings

### ISSUE THREAD

**Issue by ThomasBreuer** [Nemocas/Nemo.jl#1289]
I am not sure whether the following is a Nemo issue or an Oscar issue.
With Oscar loaded, the following works.
```
R = ResidueRing(ZZ, 2);
for a in R println(a); end
```
With Nemo loaded but Oscar not loaded, I get the following error message.
```
ERROR: MethodError: no method matching iterate(::Nemo.NmodRing)
```
With Oscar loaded, I get the following.
```
julia> R = ResidueRing(ZZ, fmpz(2));

julia> for a in R print(a); end
ERROR: MethodError: no method matching iterate(::Nemo.FmpzModRing)
[...]
```

**Comment by wbhart** [comment]
I imagine someone implemented the iterator somewhere other than Nemo, e.g. Hecke. This gets loaded by Oscar but because it is not in Nemo itself it does not work from Nemo.

There are *many* functions like this.

**Comment by wbhart** [comment]
Ah I see. I had overlooked that you had two examples, one for ZZ and one for Int.

Yes, I agree that they should be implemented in the same place. The canonical place would be Nemo in my opinion, as it benefits the greatest number of people.

But the job of moving all those functions that were implemented elsewhere into Nemo is too large for one person. I did start working on it some time back, but it isn't an easy job to do well and I don't have plans to return to it.
